homework/src/app.js

```diff
@@ -1,11 +1,184 @@
-{
-  /* Your homework goes here */
   
   /* STRINGS */
 
+  console.log(`------ STRINGS -------`);
+  console.log(`\n`);
+
+  const myString = "hello,this,is,a,difficult,to,read,sentence";
+  console.log(myString);
+  console.log(`The length of myString is: ${myString.length}`);
+
+  const myNewString = myString.replace(/,/g , ` `); // g(global) is used to replace all occurances of ',' with ' '
+  console.log(myNewString);
+  console.log(`\n`);
+
+  const str = 'dlroW olleH';
+  console.log(`Old string: ${str}`);
+
+  const arr = str.split(``);
+  arr.reverse();
+  const newStr = arr.toString();
+  const reversedStr = newStr.replace(/,/g , ``);
+  console.log(`Reversed String: ${reversedStr}`);
+  console.log(`\n`);
+
+
   /* ARRAYS */
 
+  console.log(`------ ARRAYS -------`);
+  console.log(`\n`);
+
+  const favoriteAnimals = ["blowfish", "capricorn", "giraffe"];
+  favoriteAnimals.push(`turtle`);
+  /* I thinks the new array will be favoriteAnimals = ["blowfish", "capricorn", "giraffe", "turtle"]
+  since arrays which are complex data types are mutable thus their values can be changed even if they are const. */
+  console.log(favoriteAnimals);
+
+  favoriteAnimals.splice(1, 0, "fox");
+  /* the new array will be favoriteAnimals = ["blowfish", "fox", "capricorn", "giraffe", "turtle"] as I inserted "fox"
+  at index 1 */
+  console.log(favoriteAnimals);
+
+  favoriteAnimals.splice(3, 1);
+  console.log(favoriteAnimals);
+  console.log(`\n`);
+
   /* OPERATORS */
 
-  /* QUESTIONS FROM HYF REPOS */
-}
\ No newline at end of file
+  console.log(`------ OPERATORS -------`);
+  console.log(`\n`);
+
+  /* 1 */
+
+  0 == '0'; // true
+  0 == [];  // true
+  "0" == [];  // false
+  /* The above snippet is false because seemingly different values equate to true when compared with ==  
+  (loose or abstract equality) as JavaScript converts each to a string representation before comparison.
+   "0" is a string with value 0 whereas [] represents empty array. Also there is a special rule about 
+   comparison of []. When it is compared with a truthy, in this case "0" the result is false even though 
```

Comment:
this one is tricky, `[]` gets coerced into a string, so this expression ends up being `"0" == ""` which is `false`.

Reply:
Thank you very much for the explanation.